pilot/pkg/serviceregistry/serviceentry/servicediscovery_test.go

```diff
@@ -866,6 +866,29 @@ func TestServiceDiscoveryWorkloadInstance(t *testing.T) {
 		expectServiceInstances(t, sd, selector, 0, instances)
 		expectEvents(t, events, Event{kind: "eds", host: "selector.com", namespace: selector.Namespace, endpoints: 2})
 	})
+
+	t.Run("workload instance with port map", func(t *testing.T) {
+		// Delete the instance
+		callInstanceHandlers([]*model.WorkloadInstance{fi1}, sd, model.EventDelete, t)
+		instances := make([]*model.ServiceInstance, 0)
+		expectServiceInstances(t, sd, selector, 0, instances)
+		expectProxyInstances(t, sd, instances, "2.2.2.2")
+		expectEvents(t, events, Event{kind: "eds", host: "selector.com", namespace: selector.Namespace, endpoints: 0})
+
+		// Add a workload instance, and use port map
+		fi1.PortMap = map[string]uint32 {"tcp-444": 4444, "http-445": 4445}
```

Comment:
If we modify `fi1` deep copy it since other tests could use it in the future

Reply:
The PortMap will be reset to nil by the `defer func()` in the next line.